src/hotspot/share/gc/g1/g1FullGCCompactTask.cpp

```diff
@@ -82,13 +89,12 @@ void G1FullGCCompactTask::compact_region(HeapRegion* hr) {
   assert(!hr->is_humongous(), "Should be no humongous regions in compaction queue");
 
   if (!collector()->is_free(hr->hrm_index())) {
+    // The compaction closure not only copies the object to the new
+    // location, but also clears the bitmap for it. This is needed
+    // for bitmap verification as well as for being able to use the
+    // prev_bitmap for evacuation failures.
```

Comment:
Instead of clearing the bit for each obj individually, I wonder if clearing the bitmap for the whole region will be faster. IOW, just call `collector()->mark_bitmap()->clear_region(hr);` unconditionally.

Reply:
I assume mem write is more expensive (or at least not more efficient) than mem read, so the performance changes make sense to me.

And a comment about the observation of the regression will be helpful in the future for others who think this might be a optimization point.

Reply:
Added a few lines about this.